numpy/core/arrayprint.py

```diff
@@ -161,6 +165,8 @@ def set_printoptions(precision=None, threshold=None, edgeitems=None,
     if edgeitems is not None:
         _summaryEdgeItems = edgeitems
     if precision is not None:
+        if type(precision) is not int:
```

Comment:
Alternatively, combine this line and the line above:
`if precision is not None`
into:
`if isinstance(precision, int):`